code-challenges.js

```diff
@@ -64,12 +194,54 @@ var refactorTester3 = "IAmACodingMaster"
 var utensils = ["fork", "knife", "tongs", "chopsticks", "skewer"]
 var vessels = ["cup", "mug", "stemware", "glass", "tumbler", "tea cup"]
 
-
-
+//create a function called longestString
+//takes in an array of strings
+const longestString = (array) => {
+    //create a variable that stores the lengths of each string in each array
+    let max = array.length
+    // combining the map and math.max functions to iterate though each array and find the length of each
+    array.map(value => max = Math.max(max, value.length))
+    //using filter to return a new array with the longest string out of the previous array
+    return array.filter(value => value.length === max)
+  }
+  
+console.log(longestString(utensils))
+console.log(longestString(vessels))
 
 
 // --------------------7) Create a function that takes in an array of strings and returns an array with the last letter removed from each string.
 // Expected output: ["Luig", "Peac", "Toa", "Yosh"] ["Ink", "Blink", "Pink", "Clyd"]
 
 var mario = ["Luigi", "Peach", "Toad", "Yoshi"]
 var pacman = ["Inky", "Blinky", "Pinky", "Clyde"]
+
+//!!!!!!! attempt one !!!!! the uncommented one is when I used a .map instead to tighten up code
+//create a function called removeLast
+//takes in one argument that is an array
+// const removeLast = (array) => {  
+//     // create a new array to store the strings that had last letter removed <- aha moment during pseudo
+//     let lastLetterArr = []
+        // iterate trough each value in the array
+//     for(let i = 0 ; i<array.length; i++){
+            //add each value from original array to new array with the last letter taken off using slice
+//         lastLetterArr.push(array[i].slice(0 ,-1))
+//     }
+    //returns the new array that stores the new stings sans last letter
+//  return lastLetterArr
+// }
+// console.log(removeLast(mario))
+// console.log(removeLast(pacman)) 
+
+//---7 attempt 2 !!!(use this one)!!!
+
+//create a function named removeLast
+// takes in one argument that is an array
+const removeLast = (array) => {
+    //create a variable that will store the mapped array and also return it when function is called
+    return mappedArray = array.map(value => {
+        //return the result of iterating though each sting and removing the last character using slice
+        return value.slice(0 ,-1)
```

Comment:
clever use of  the `-1` index. Though, this won't always be obvious to people what `-1` is doing.